File: cpp/medium/reverse.cpp
#include <iostream>
#include <string>
#include <algorithm>

class Manipulate {
public:
    Manipulate(const std::string& text) : text(text) {}

    std::string reverse() {
        int l = 0, r = text.size() - 1;
        while (l < r) {
            std::swap(text[l], text[r]);
            l++;
            r--;
        }
        return text;
    }

private:
    std::string text;
};

int main() {
    Manipulate string_manipulator("hello");
    std::string reversed_text = string_manipulator.reverse();
    std::cout << reversed_text << std::endl;
    return 0;
}

File: dsa/reverse.py
class Manipulate: 
    def __init__(self, text):
        self.text = text
    

    def reverse(self):
        text_list = list(self.text)
        l , r = 0, len(text_list) - 1
        while l < r: 
            text_list[l], text_list[r] = text_list[r], text_list[l]
            l += 1
            r -= 1
        self.text = "".join(text_list)
        return self.text


if __name__ == "__main__":
    string_manipulator = Manipulate("hello")
    reversed_text = string_manipulator.reverse()
    print(reversed_text)
